editor/src/document/document_message_handler.rs

```diff
@@ -43,6 +44,52 @@ impl DocumentsMessageHandler {
 	pub fn active_document_mut(&mut self) -> &mut DocumentMessageHandler {
 		&mut self.documents[self.active_document_index]
 	}
+	fn get_new_doc_title(&self) -> String {
+		let digits = ('0'..='9').collect::<Vec<char>>();
+		let mut doc_title_numbers = self
+			.documents
+			.iter()
+			.map(|d| {
+				if d.name.ends_with(digits.as_slice()) {
+					let (_, number) = d.name.split_at(17);
+					number.trim().parse::<usize>().unwrap()
+				} else {
+					1
+				}
+			})
+			.collect::<Vec<usize>>();
+		doc_title_numbers.sort_unstable();
+		let mut new_doc_title_num = 1;
+		while new_doc_title_num <= self.documents.len() {
+			if new_doc_title_num != doc_title_numbers[new_doc_title_num - 1] {
+				break;
+			}
+			new_doc_title_num += 1;
+		}
+		let name = match new_doc_title_num {
+			1 => "Untitled Document".to_string(),
```

Comment:
"Untitled Document" should probably be moved to a constant that is also used in the name matching